tt_metal/llrt/tt_cluster.cpp

```diff
@@ -341,6 +341,12 @@ uint32_t Cluster::get_harvested_rows(chip_id_t chip) const {
     }
 }
 
+eth_coord_t Cluster::get_chip_location(chip_id_t chip) const {
+    static std::unordered_map<chip_id_t, eth_coord_t> chip_locations = this->cluster_desc_->get_chip_locations();
```

Comment:
I'm also thinking that typically the call to `get_chip_location` is only something an app (like ttrt) will do offline and infrequently so not too worried about overhead.

Reply:
Hey @nsmithtt great point, @aliuTT brought this up too - it's not expensive so I'll just initialize it as a local variable. @aliuTT also suggested we should directly query this cluster API instead of adding it to device. Pushed an update that does that as well.